sklearn/metrics/_regression.py

```diff
@@ -1301,12 +1301,14 @@ def _mean_tweedie_deviance(y_true, y_pred, sample_weight, power):
     """Mean Tweedie deviance regression loss."""
     xp, _ = get_namespace(y_true, y_pred)
     p = power
+    zero = xp.asarray(0, dtype=y_true.dtype)
     if p < 0:
         # 'Extreme stable', y any real number, y_pred > 0
         dev = 2 * (
-            xp.pow(xp.where(y_true > 0, y_true, 0), 2 - p) / ((1 - p) * (2 - p))
-            - y_true * xp.pow(y_pred, 1 - p) / (1 - p)
-            + xp.pow(y_pred, 2 - p) / (2 - p)
+            xp.pow(xp.where(y_true > 0, y_true, zero), xp.asarray(2 - p))
```

Comment:
I am surprise that you don't need to pass the device and dtype of `y_true` or `y_pred` to those `xp.asarray` calls. But the tests did pass on cuda... There must be a tolerance for scalar arrays.